source_code/main_mcu/src/LOGIC/logic_encryption.c

```diff
@@ -22,15 +22,31 @@
 #include <string.h>
 #include "logic_encryption.h"
 #include "bearssl_block.h"
+#include "bearssl_hash.h"
+#include "bearssl_hmac.h"
+#include "bearssl_rand.h"
+#include "bearssl_ec.h"
 #include "custom_fs.h"
 #include "nodemgmt.h"
+#include "rng.h"
+
 // Next CTR value for our AES encryption
 uint8_t logic_encryption_next_ctr_val[MEMBER_SIZE(nodemgmt_profile_main_data_t, current_ctr)];
 // Current encryption context */
 br_aes_ct_ctrcbc_keys logic_encryption_cur_aes_context;
 // Current user CPZ user entry
 cpz_lut_entry_t* logic_encryption_cur_cpz_entry;
 
+static br_sha256_context logic_encryption_sha256_ctx;
```

Comment:
quick comment to explain how/where these are used?

Reply:
sure